administrator/components/com_fields/models/fields/type.php

```diff
@@ -91,4 +107,31 @@ protected function getOptions ()
 
 		return $options;
 	}
+
+	/**
+	 * Parses the file with the given path. If it is a class starting with the
+	 * name JFormField and implementing JFormDomfieldinterface, then the class name is returned.
+	 *
+	 * @param string $path
+	 * @return string|boolean
+	 */
+	private function getClassNameFromFile($path)
+	{
+		$tokens = token_get_all(JFile::read($path));
```

Comment:
If it's just checking it implements the interface why don't you just use php's instanceof function (http://php.net/manual/en/language.operators.type.php - see example 4 here)